python/GangaLHCb/Lib/Applications/BenderBox.py

```diff
@@ -0,0 +1,521 @@
+    layout    = """#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# =============================================================================
+importOptions('{datafile}')
+from Gaudi.Configuration import EventSelector,FileCatalog
+from copy import deepcopy
+inputdata    = deepcopy( EventSelector() .Input    )
+filecatalogs = deepcopy( FileCatalog  () .Catalogs )
+EventSelector().Input    = [] ## reset it 
+FileCatalog  ().Catalogs = [] ## reset it 
+# =============================================================================
+# configure Bender and run it 
+# =============================================================================
+import  sys
+sys.path = sys.path + ['.']
+import {modulename} as USERMODULE
+USERMODULE.configure( inputdata , filecatalogs {paramstring})
+USERMODULE.run({events})
+# =============================================================================
+# The END
+# =============================================================================
+"""
+
+    ## Returns file names which should be copyed to shared area
+    def inputfiles(self):
+        "Returns file names which should be copyed to shared area"
+        return [LocalFile(self.module.name),]
+
+    ## Returns  wrapper file which run at WN
+    def getWrapper(self, data_file):
+        " Returns  wrapper file which run at WN"
+        
+        module_name  = os.path.split(self.module.name)[-1].split('.')[0]
+        param_string =  ',params=%s' % self.params if self.params else ''
+        
+        the_script      = self.layout.format (
+            datafile    = data_file ,
+            modulename  = module_name ,
+            paramstring = param_string ,
+            events      = self.events
+            )
+
+        from GangaLHCb.Lib.RTHandlers.RTHUtils import getXMLSummaryScript
+        the_script += getXMLSummaryScript()
+
+        
+        return the_script
+
+
+# =============================================================================
+## @class bender
+#  Helper class to define the main proeprties of BenderScript application
+#  - the scripts to be executed 
+#  - the configuration scripts (aka ``options'') to be imported 
+#  - bender commands to be executed
+#  - other arguments
+#
+#  The scripts are executed within ``bender'' context
+#  The configuration scripts (aka ``options'') are ``imported'' using <code>importOptions</code> costruction
+# 
+#  The actual command to be executed is:
+#  @code 
+#  > bender [ scripts [ scripts ...  --no-color [ arguments ] --import [ imports [ imports [ ... --no-castor --import=data.py --batch   [ --command  [ commands [ ... 
+#  @endcode
+#  Usage:
+#  @code 
+#  j.application = BenderBox ( tool = bender ( scripts   = ['path_to_script/the_script.py']  ,
+#                                              commands  = [ 'print ls()' ]  ) )
+#  @endcode 
+#  @author Vladimir ROMANOVSKY Vladimir.Romanovskiy@cern.ch
+#  @author Vanya BELYAEV  Ivan.Belyaev@itep.ru       
+class bender(GangaObject):
```

Comment:
There should be no lowercase class names. This is in particular confusing with the `Bender` GPI name of the legacy `Bender` application type.